### PR TITLE
Change md5withRSA to SHA1withRSA which is the default sig alg.

### PR DESCRIPTION
**Pull request by fekberg**
SHA1withRSA is now the default signature algorithm according to this JDK issue: https://bugs.openjdk.java.net/browse/JDK-6560733

### CODE REVIEW

**Comment by fekberg** [comment]
I'd much rather see `apksigner` being used instead as per this https://bugzilla.xamarin.com/show_bug.cgi?id=57914

Meanwhile, I think moving away from MD5 is a good idea.

**Comment by fekberg** [comment]
Do you have any records on what those devices are? I've got a hard time finding any good information about it.

Parameterising `-sigalg` is probably a better idea than changing it to another default value.

**Comment by dellis1972** [comment]
@fekberg I have an old Galaxy S1 which is on Android 2.3.6 which won't installer apps signed with a new algorithm. You get a weird error from google play when you try to. Its been a while though so I can't remember the exact problem.

**Comment by fekberg** [comment]
@dellis1972 That's a very old phone.. :) I haven't seen any apps that still targets anything below 4.x

**Comment by dellis1972** [comment]
I know games that target 2.3.6, its a very small part of the market though. Still, doing this with a parameter that the developer can override is probably the best way forward in this case. 

I'm gonna work on `apksigner` this week so hopefully that will help too.

**Comment by fekberg** [comment]
@dellis1972 We had a discussion yesterday on Slack in regards of adding `apksigner` instead. Not sure you saw that?

Anyways, in the meantime, I could update the PR to parameterize this instead. Haven't done that before, is it enough to just add it as a property in this class and it magically works?

**Comment by dellis1972** [comment]
you can add a new string property `SigningAlgorithm` or something to the class. It will not need the `[Required]` attribute since we will make it optional. 
The you can do a null check and replace the value with the old default `md5withRSA`. 

You will also need to edit `Xamarin.Android.Common.targets`. Look for the `_Sign` target and add 

`SigningAlgorithm="$(AndroidSigningAlgorithm)"`

at around [1].


[1] https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L2441.

**Comment by dellis1972** [comment]
if you are feeling adventurous you could also document it at some point around https://github.com/xamarin/xamarin-android/blob/master/Documentation/build_process.md#AndroidVersionCodeProperties :P

**Comment by fekberg** [comment]
Awesome, thanks for pointing these out!

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -2439,6 +2439,7 @@ because xbuild doesn't support framework reference assemblies.
 		ToolExe="$(JarsignerToolExe)"
 		TimestampAuthorityUrl="$(JarsignerTimestampAuthorityUrl)"
 		TimestampAuthorityCertificateAlias="$(JarsignerTimestampAuthorityCertificateAlias)"
+		SigningAlgorithm="$(_ApkSigningAlgorithm)"
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets]: we should use `ApkSigningAlgorithm` rather than `_ApkSigningAlgorithm`. In msbuild the convention is `_` prefixed properties are private and should not be used by the end user. Ones without are public. So we should remove the `_` since we expect the user to override this.

**Comment by fekberg** [comment]
Should be corrected now!

**Comment by dellis1972** [comment]
build

**Comment by dellis1972** [comment]
@fekberg cool. perfect. Just waiting for the PR builder to give it the thumbs up and we are good :) thanks for the PR!

File: src/Xamarin.Android.Build.Tasks/Tasks/AndroidSignPackage.cs [hunk 2]
```diff
@@ -42,7 +44,7 @@ protected override string GenerateCommandLineCommands ()
 			cmd.AppendSwitchIfNotNull ("-storepass ", StorePass);
 			cmd.AppendSwitchIfNotNull ("-keypass ", KeyPass);
 			cmd.AppendSwitchIfNotNull ("-digestalg ", "SHA1");
-			cmd.AppendSwitchIfNotNull ("-sigalg ", "md5withRSA");
+			cmd.AppendSwitchIfNotNull ("-sigalg ", string.IsNullorWitespace(SigningAlgorithm) ? "md5withRSA" :SigningAlgorithm);
```
Comment (dellis1972) [inline src/Xamarin.Android.Build.Tasks/Tasks/AndroidSignPackage.cs]: @fekberg we have a typo here

`IsNullorWitespace` vs `IsNullOrWitespace`

Comment (fekberg) [reply src/Xamarin.Android.Build.Tasks/Tasks/AndroidSignPackage.cs]: Oh my, sorry about that!

**Comment by dellis1972** [comment]
@fekberg really sorry, looks like you committed changes to the .sln.

**Comment by fekberg** [comment]
I was sure I reverted that.. Let me try again :)

**Comment by dellis1972** [comment]
cool :) thanks

**Comment by dellis1972** [comment]
build

**Comment by dellis1972** [comment]
build

**Comment by fekberg** [comment]
@dellis1972 Thanks for all the help, happy I can be of service!

**Comment by jonpryor** [comment]
Manually merged, and documented added.
